src/config.js

```diff
@@ -1049,16 +1095,20 @@ class Config {
         this[settingsToChange] = value
       }
     }
-    return this.emitChangeEvent()
+    return this.emitChangeEvent() && this.emitUpdateEvent(value)
   }
 
   observeKeyPath (keyPath, options, callback) {
     callback(this.get(keyPath))
     return this.onDidChangeKeyPath(keyPath, event => callback(event.newValue))
   }
 
+  listenKeyPath (keyPath, options, callback) {
```

Comment:
This function name doesn't really make sense as far as readability is concerned. Essentially all other functions in this class read like English, but this function does not.

A better name might be something along the lines of `listenForKeyPath`.